dcm4che-imageio/src/main/java/org/dcm4che3/imageio/plugins/dcm/DicomImageReader.java

```diff
@@ -161,23 +165,26 @@ public void setInput(Object input, boolean seekForwardOnly,
             }
         } else if (input instanceof DicomMetaData) {
             DicomMetaData metadata = (DicomMetaData) input;
-            initPixelData(metadata.getAttributes());
+            initPixelDataFromAttributes(metadata.getAttributes());
             initPixelDataFile();
             setMetadata(metadata);
         } else {
             iis = (ImageInputStream) input;
         }
     }
 
-    private void initPixelData(Attributes ds) {
+    private void initPixelDataFromAttributes(Attributes ds) {
         VR.Holder holder = new VR.Holder();
         Object value = ds.getValue(Tag.PixelData, holder);
         if (value != null) {
             pixelDataVR = holder.vr;
             if (value instanceof BulkData) {
                 pixelData = (BulkData) value;
                 pixelDataLength = pixelData.length();
-            } else { // value instanceof Fragments)
+            } else if( value instanceof byte[] ) {
```

Comment:
This is needed to handle decoding uncompressed full Attributes instance data, added an integration test for this type of data.